docs/jsx.md

```diff
@@ -134,4 +136,82 @@ You're effectively passing the array `[| <div />, <div /> |]` to `MyReasonCompon
 <MyReasonComponent> <div /> </MyReasonComponent>
 ```
 
-We unwrap this for you automatically to just `<div />` instead of an array of a single element.
\ No newline at end of file
+We unwrap this for you automatically to just `<div />` instead of an array of a single element.
+
+## Migrating from version 2 to version 3
+
+There are many ReasonReact applications, so it is hard to define "The One True" migration strategy for them all.
+
+Depending on the size and nature of your application there are two options available to migrate from JSX version 2 to version 3.
+
+### Application level
+
+By adding `{"reason": {"react-jsx": 3}` in your [`bsconfig.json`](https://bucklescript.github.io/docs/en/build-configuration.html#reason-refmt).
+
+This approach requires that all components in the application must be made compatible with version 3 of JSX at once, so it will be a better fit for smaller apps with a reduced number of components, where all of them can be migrated to version 3 in one fell swoop.
+
+### File level
+
+For larger applications, it might not be possible to migrate all components at once. In these cases, a per-file migration is also possible.
+
+A file can be configured to use version 3 of the transform by adding `[@bs.config {jsx: 3}]` at the top of the file.
+
+The per-file configuration allows to mix, in the same application, components compatible with either of both versions of the JSX transforms. However, the restriction is that all the components used in a file will have to be compatible with the JSX version specified for that file.
+
+For example, if a file contains the following code:
+
+```reason
+/* User.re */
+[@bs.config {jsx: 3}]
```

Comment:
I think `;` is missing at the end of the line here 🙂

Reply:
@baransu Oh, you're right, thanks! That's why I was getting warnings from the compiler 😅 fixed!